components/script/dom/request.rs

```diff
@@ -341,7 +346,19 @@ impl Request {
         }
 
         // Step 31
-        try!(r.Headers().fill(Some(HeadersInit::Headers(headers_copy))));
+        if let Some(HeadersInit::Headers(_)) = init.headers {
+            try!(r.Headers().fill(Some(HeadersInit::Headers(headers_copy))));
+        };
+
+        // This is equivalent to the specification's concept of
+        // "associated headers list". If an init headers is not given,
+        // but an input with headers is given, set request's
+        // headers as the input's Headers.
+        if let None = init.headers {
```

Comment:
nit: if `init.headers.is_none`, though this can probably be more legible with a match statement combined with the above `if let`.